debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/ReverseFileReader.java

```diff
@@ -0,0 +1,56 @@
+package io.debezium.server.ybexporter;
+
+import java.io.RandomAccessFile;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.io.IOException;
+
+public class ReverseFileReader implements AutoCloseable {
+    private MappedByteBuffer buffer;
+    private long currentPosition;
+
+    public ReverseFileReader(String filePath) throws IOException {
+        try (RandomAccessFile raf = new RandomAccessFile(filePath, "r");
+                FileChannel channel = raf.getChannel()) {
+            long fileSize = channel.size();
+            this.buffer = channel.map(FileChannel.MapMode.READ_ONLY, 0, fileSize);
+            this.currentPosition = fileSize - 1;
+        }
+    }
+
+    public Boolean hasNext() {
+        return currentPosition >= 0;
+    }
+
+    public String nextLine() {
```

Comment:
Let's not do this: reading file in reverse is confusing and not performant (while reading in reverse order the read-ahead mechanism in the kernel won't work).
Instead, let's go with what Aneesh suggested: the `EventQueue` passes ordered list of `EventQueueSegments` to the `EventDedupeCache`. The `warmUp()` method, reads the events in-order oldest-to-newest and populates the cache.